lib/stack_master/commands/apply.rb

```diff
@@ -87,14 +97,43 @@ def create_stack
         cf.create_stack(stack_options.merge(tags: proposed_stack.aws_tags))
       end
 
+      def upload_files
+        s3.upload_files(s3_options)
+      end
+
+      def template_method
+        return :template_body unless use_s3?
+        :template_url
+      end
+
+      def template_value
+        return proposed_stack.maybe_compressed_template_body unless use_s3?
+        s3.url(@s3_config.merge('template' => @stack_definition.template))
+      end
+
+      def files_to_upload
+        return {} unless use_s3?
+        {@stack_definition.template => @stack_definition.template_file_path}
```

Comment:
This won't work for anything but json templates. We need to ensure the compiled template is sent for sparkleformation/.rb templates. The method to return a compiled string is currently `proposed_stack.maybe_compressed_template_body`. Also, we probably don't want to use the `.rb` extension for the file on S3 given it would be JSON at that stage.